2018 edition

lespea opened the pull request:
Updated the dependencies and switched to the 2018 edition.  You might want to look at the "Fix failing test" commit as I wasn't sure exactly what was going on there.

BurntSushi commented on .travis.yml:
```diff
@@ -18,10 +18,10 @@ matrix:
       env: TARGET=x86_64-unknown-linux-gnu
     - os: linux
-      rust: 1.28.0
+      rust: 1.32.0
```
I would just go ahead and bump this up to 1.34.0.

lespea replied:
Okay will do

BurntSushi commented on Cargo.toml:
```diff
@@ -29,13 +30,13 @@ opt-level = 3
+crossbeam-channel = "0.3"
 csv = "1"
-csv-index = "0.1.5"
+csv-index = "0.1"
```
This should probably remain as `0.1.5` (unless you've specifically tested that `xsv` can work with `0.1.0`).

lespea replied:
I was just going off of semver that it should be fine... but I can specify the exact version if you want.  I didn't do any special testing or anything.

BurntSushi commented on src/cmd/split.rs:
```diff
@@ -116,11 +113,9 @@ impl Args {
         }
-        drop(tx);
-        rx.recv();
+        pool.join();
```
Nice simplification!

BurntSushi commented on src/util.rs:
```diff
@@ -8,14 +8,14 @@ use std::time;
 
 pub fn num_cpus() -> usize {
-    num_cpus::get()
+    ncpus::get()
```
Why not just use `num_cpus` here? The renaming doesn't seem worth it.

lespea replied:
There was a compiler error with 2018 over conflicting names.  Tbh I didn't look if there was a better way of resolving it so I just renamed the import.  I'll look to see if there's a better way.

lespea commented:
Okay let me know if you'd like anything else changed.

lespea commented:
Is there anything else you need for this to get merged?

bronson commented:
nudge?

lespea commented:
Update the libraries again to the latest versions, cleaned up the imports + unneeded lifetimes, fixed (or disabled) the remaining clippy lints.  Had to bump the version to 1.36.0 for some of the library updates, but that version is pretty old by now so didn't think it would be an issue.

Let me know if you need me to do anything else for this to be merged...
